### PR TITLE
Updated all pieces for 0.5.0

### LINKED ISSUE #20: Feature Request: PostgreSQL Provider
I would like someone to help expand our official plugins to add support for PostgreSQL databases. The Provider should of course be SGv2 compatible, as even those who are using klasa v0.4.0 will be able to use the up to date provider. (SGv2 Providers are backwards compatible with SGv1)

For whoever wants to pick this up, I had started on writing this myself; although not very far into it:

```js
const { Provider } = require('klasa');
const { Pool } = require('pg');

module.exports = class PostgreSQL extends Provider {

	constructor(...args) {
		super(...args, {
			enabled: true,
			sql: true,
			description: 'Allows you to use PostgreSQL functionality throught Klasa'
		});
		this.TYPES = DATATYPES;
		this.db = null;
	}

	init() {
		this.db = new Pool({
			host: 'localhost',
			port: 5432,
			user: 'database-user',
			password: 'database-password',
			database: 'Klasa',
			max: 20,
			idleTimeoutMillis: 30000,
			connectionTimeoutMillis: 2000
		});

		this.db.on('error', err => this.client.emit('error', err));
	}

	shutdown() {
		return this.db.end();
	}
```

With this, most things should be `const someValue = await this.db.query(someQuery);` as PG returns promises, and using the Pool constructor, it maintains and can self manage client checkouts to make transactions to the database, rather than creating a new client for every database transaction. Docs: https://node-postgres.com/

### PR DESCRIPTION
**Pull request by kyranet**
Fixes #20

### CODE REVIEW

**Comment by MrJacz** [comment]
MongoDB provider 🤔

**Comment by kyranet** [comment]
When Cyber decides to fix it.

**Comment by MrJacz** [comment]
I fixed the hasTable function what else is wrong with it?

**Comment by kyranet** [comment]
It lacks of the new methods

**Comment by MrJacz** [comment]
I see

File: extendables/advchannel.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+const { Extendable } = require('klasa');
+const { Channel, Message } = require('discord.js');
+const { regExpEsc } = require('../util/util');
```
Comment (MrJacz) [inline extendables/advchannel.js]: this is actually klasa not Skyra so util#regExpEsc

File: extendables/advrole.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
+const { Extendable } = require('klasa');
+const { Role } = require('discord.js');
+const { regExpEsc } = require('../util/util');
```
Comment (MrJacz) [inline extendables/advrole.js]: same here

File: extendables/advuser.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
+const { Extendable } = require('klasa');
+const { GuildMember, User } = require('discord.js');
+const { regExpEsc } = require('../util/util');
```
Comment (MrJacz) [inline extendables/advuser.js]: same here

**Pull request merged**